packages/hdwallet-walletconnect/src/walletconnect.ts

```diff
@@ -0,0 +1,317 @@
+    }
+  }
+
+  public ethNextAccountPath(_msg: core.ETHAccountPath): core.ETHAccountPath | undefined {
+    return undefined;
+  }
+
+  public async ethSupportsNetwork(chainId = 1): Promise<boolean> {
+    return chainId === 1;
+  }
+
+  public async ethSupportsSecureTransfer(): Promise<boolean> {
+    return false;
+  }
+
+  public ethSupportsNativeShapeShift(): boolean {
+    return false;
+  }
+
+  public async ethSupportsEIP1559(): Promise<boolean> {
+    return false;
+  }
+
+  public ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+    const slip44 = core.slip44ByCoin(msg.coin);
+    if (slip44 === undefined) return [];
+    return [
+      {
+        addressNList: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx, 0, 0],
+        hardenedPath: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx],
+        relPath: [0, 0],
+        description: "WalletConnect",
+      },
+    ];
+  }
+}
+
+export class WalletConnectHDWallet implements core.HDWallet, core.ETHWallet {
+  readonly _supportsETH = true;
+  readonly _supportsETHInfo = true;
+  readonly _supportsBTCInfo = false;
+  readonly _supportsBTC = false;
+  readonly _isWalletConnect = true;
+
+  info: WalletConnectWalletInfo & core.HDWalletInfo;
+  provider: WalletConnectProvider;
+  connected = false;
+  chainId: number = -1;
+  accounts: string[] = [];
+  ethAddress: string = "";
+
+  constructor(provider: WalletConnectProvider) {
+    this.provider = provider;
+    this.info = new WalletConnectWalletInfo();
+  }
+
+  async getFeatures(): Promise<Record<string, any>> {
+    return {};
+  }
+
+  public async isLocked(): Promise<boolean> {
+    return false;
+  }
+
+  public getVendor(): string {
+    return "WalletConnect";
+  }
+
+  public async getModel(): Promise<string> {
+    return "WalletConnect";
+  }
+
+  public async getLabel(): Promise<string> {
+    return "WalletConnect";
+  }
+
+  /**
+   * Initialize
+   * 
+   * Subscribes to EIP-1193 events
+   */
+  public async initialize(): Promise<void> {
+    await this.provider.enable()
+
+    if (this.provider.connector.connected) {
+      const { chainId, accounts } = this.provider.connector;
+      const [address] = accounts;
+      this.connected = true;
+      this.chainId = chainId;
+      this.accounts = accounts;
+      this.ethAddress = address;
+    }
+  }
+
+  public hasOnDevicePinEntry(): boolean {
+    return this.info.hasOnDevicePinEntry();
+  }
+
+  public hasOnDevicePassphrase(): boolean {
+    return this.info.hasOnDevicePassphrase();
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return this.info.hasOnDeviceDisplay();
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return this.info.hasOnDeviceRecovery();
+  }
+
+  public hasNativeShapeShift(srcCoin: core.Coin, dstCoin: core.Coin): boolean {
+    return this.info.hasNativeShapeShift(srcCoin, dstCoin);
+  }
+
+  public supportsOfflineSigning(): boolean {
+    return false;
```

Comment:
If `this.provider.wc.signTransaction(msg)` does not also broadcast the signed tx, then this should be set to `true`.

Reply:
Thank you for this detail!

Reply:
It always sends the request in signTransaction.